eclipse-collections/src/main/java/com/fasterxml/jackson/datatype/eclipsecollections/deser/pair/PairInstantiators.java

```diff
@@ -94,107 +94,137 @@ static void add(Class<?> objectKeyOrValuePairClass,
         KEY_OR_VALUE_OBJECT_LAMBDAS.put(objectKeyOrValuePairClass, lambda);
     }
 
+    /**
+     * primitive->object
+     */
     @SuppressWarnings("unused") // Used from PairInstantiatorsPopulator
     static <P> ValueInstantiator primitiveObjectInstantiator(
             JavaType inputType, Class<?> one,
             BiFunction<Object, Object, P> factory
     ) {
-        return new ValueInstantiator.Base(inputType) {
+        return new PairInstantiator(inputType) {
             @Override
-            public boolean canCreateFromObjectWith() {
-                return true;
+            public Object createFromObjectWith(DeserializationContext ctxt, Object[] args) {
+                return factory.apply(args[0], args[1]);
             }
 
             @Override
-            public SettableBeanProperty[] getFromObjectArguments(DeserializationConfig config) {
-                JavaType oneType = config.constructType(one);
-                JavaType twoType = inputType.containedType(0);
-                return makeProperties(config, oneType, twoType);
+            JavaType oneType(DeserializationConfig config) {
+                return config.constructType(one);
             }
 
             @Override
-            public Object createFromObjectWith(DeserializationContext ctxt, Object[] args) {
-                return factory.apply(args[0], args[1]);
+            JavaType twoType(DeserializationConfig config) {
+                return inputType.containedType(0);
```

Comment:
This works, although for extra safety could also use `containedTypeOrUnknown()` (which works even if no parameters were resolved). Might hide issues tho so could use this for now and only resort to safer alternative if shown to be needed.

Reply:
yea containedTypeOrUnknown is better or it'll end up like in #75. I'm replacing the check in #75 with containedTypeOrUnknown as well since it does the same thing